### PR TITLE
Add Nginx 1.21.4 support

### PR DESCRIPTION
**Pull request by julek-wolfssl**
Depends on https://github.com/wolfSSL/wolfssl/pull/4610

### CODE REVIEW

File: nginx-1.21.4-wolfssl.patch [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
++
+ ngx_int_t
+ ngx_ssl_client_certificate(ngx_conf_t *cf, ngx_ssl_t *ssl, ngx_str_t *cert,
+     ngx_int_t depth)
+@@ -1024,7 +1041,7 @@ ngx_ssl_verify_callback(int ok, X509_STORE_CTX *x509_store)
+         subject = X509_NAME_oneline(sname, NULL, 0);
+         if (subject == NULL) {
+             ngx_ssl_error(NGX_LOG_ALERT, c->log, 0,
+-                          "X509_NAME_oneline() failed");
++                          "X509_NAME_oneline() failed1");
```
Comment (SparkiDev) [inline nginx-1.21.4-wolfssl.patch:121]: Doesn't look necessary.

File: nginx-tests-patches/2021-11-23-2786c65025940da977a65009e18aa3c5d23a92a6.patch [hunk 2]
```diff
@@ -0,0 +1,5744 @@
...
+ Happy testing!
+diff --git a/access.t b/access.t
+index ab26ec0..af3ea5b 100644
+--- a/access.t
++++ b/access.t
+@@ -29,6 +29,7 @@ $t->write_file_expand('nginx.conf', <<'EOF');
+ 
+ %%TEST_GLOBALS%%
+ 
++master_process off;
```
Comment (SparkiDev) [inline nginx-tests-patches/2021-11-23-2786c65025940da977a65009e18aa3c5d23a92a6.patch:453]: What's this do in the testing?

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: nginx-1.21.4-wolfssl.patch (hunk 1 above)
Comment (julek-wolfssl) [reply nginx-1.21.4-wolfssl.patch:121]: Left over from debugging. To be removed.

File: nginx-tests-patches/2021-11-23-2786c65025940da977a65009e18aa3c5d23a92a6.patch (hunk 2 above)
Comment (julek-wolfssl) [reply nginx-tests-patches/2021-11-23-2786c65025940da977a65009e18aa3c5d23a92a6.patch:453]: This instructs Nginx to not fork itself into a master<->worker. With one process, it allows debugging through GDB. Multi-process debugging doesn't seam to work for me so this was necessary. Doesn't affect the tests except in a few cases (those do not have this instruction included).

Comment (SparkiDev) [reply nginx-tests-patches/2021-11-23-2786c65025940da977a65009e18aa3c5d23a92a6.patch:453]: I would rather keep the patch to a minimum.
Please remove this in the files except for cases that we debug a lot.

File: nginx-1.21.4-wolfssl.patch (hunk 1 above)
Comment (julek-wolfssl) [reply nginx-1.21.4-wolfssl.patch]: Removed.

File: nginx-tests-patches/2021-11-23-2786c65025940da977a65009e18aa3c5d23a92a6.patch (hunk 2 above)
Comment (julek-wolfssl) [reply nginx-tests-patches/2021-11-23-2786c65025940da977a65009e18aa3c5d23a92a6.patch]: Removed `master_process off;`.

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**